packages/teams-js/src/public/calendar.ts

```diff
@@ -106,4 +127,55 @@ export namespace calendar {
     /** The body content of the meeting. */
     content?: string;
   }
+
+  export interface JoinMeetingParams {
+    skypeTeamsMeetingUrl: string;
+    subject: string;
+    source: string;
+    conversationId: string;
+    messageId: string;
+    rootMessageId: string;
+    isBroadcastMeeting?: boolean | undefined;
+    t2BroadcastParticipantRoleDetails?: IT2BroadcastParticipantRoleDetails | undefined;
+  }
+
+  export interface IT2BroadcastParticipantRoleDetails {
+    broadcastId: string;
+    broadcastRole: Promise<BroadcastParticipantRole> | BroadcastParticipantRole;
+    isBroadcastMeeting: boolean;
+    broadcastCorrelationId: string;
+    allowBroadcastProducerJoinsInT2: boolean;
+    isT2SupportedBroadcastMeeting: (
+      broadcastRole: BroadcastParticipantRole,
+      allowBroadcastProducerJoinsInT2: boolean,
+      clientType?: ClientType,
+    ) => boolean;
+    handleBroadcastMeetingRedirect: (
+      broadcastRole: BroadcastParticipantRole,
+      entityCommand: any,
+      broadcastId: string,
+      logger: any,
+      broadcastCorrelationId: string,
+      clientType?: ClientType,
+    ) => void;
+  }
+
+  export enum BroadcastParticipantRole {
+    Attendee = 'Attendee',
+    Contributor = 'Contributor',
+    None = 'None',
+    Producer = 'Producer',
+  }
+
+  export enum ClientType {
```

Comment:
How is clientType used?
For our case, wouldn't it be OutlookWeb only?
We should confirm this.

Reply:
Removed it. Was being used for broadcast role details.